app/components/op_primer/action_menu_helpers.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module OpPrimer
+  module ActionMenuHelpers
+    private
+
+    # Intended to help in building action menus with different sections, separated by dividers. Wrap all items that
+    # should go into the same section into a divider_section block. Dividers are only rendered if needed (no empty sections).
+    #
+    # @example
+    #   def render_action_menu(menu)
+    #     divider_section { menu.with_item(label: "Item alone in section") }
+    #     divider_section do
+    #       menu.with_item(label: "hidden section") if false
+    #     end
+    #     divider_section do
+    #       menu.with_item(label: "first in section")
+    #       menu.with_item(label: "last in section")
+    #     end
+    #   end
+    def divider_section(menu)
+      pending_divider = menu.items.any? && menu.items.last[:type] != :divider
+      if pending_divider
+        menu.with_divider
+      end
+
+      size_before = menu.items.size
+      yield
+
+      # undo insertion of divider, if no items were added in this section
+      menu.items.pop if menu.items.size == size_before
```

Comment:
❓  What if no items were added AND `pending_divider` was `false`, e.g. due to being the first section. Will `pop` simply return nil and nothing happens? This is what happens on ruby arrays, right?